### PR TITLE
Tar: for directories, use a default mask that has the 'x' bit.

### PR DESCRIPTION
**Pull request by tmds**
Without the 'x'-bit, the directories that get created are not accessible.

@eerhardt @carlossanlop  @dotnet/area-system-io ptal.

### CODE REVIEW

File: src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs [hunk 1]
```diff
@@ -22,8 +22,18 @@ internal static class TarHelpers
...
-            UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.GroupRead | UnixFileMode.OtherRead;
+        private const UnixFileMode DefaultFileMode =
+            UnixFileMode.UserRead | UnixFileMode.UserWrite |
+            UnixFileMode.GroupRead |
+            UnixFileMode.OtherRead;
+
+        private const UnixFileMode DefaultDirectoryMode =
+            UnixFileMode.UserExecute | UnixFileMode.UserRead | UnixFileMode.UserWrite |
+            UnixFileMode.GroupExecute | UnixFileMode.GroupRead |
+            UnixFileMode.OtherExecute | UnixFileMode.OtherRead;
```
Comment (stephentoub) [inline src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs:33]:
```suggestion
        private const UnixFileMode DefaultDirectoryMode =
            DefaultFileMode |
            UnixFileMode.UserExecute | UnixFileMode.GroupExecute | UnixFileMode.OtherExecute;
```

**Review by stephentoub — APPROVED** [review]

**Comment by tmds** [comment]
> Do we need a test that untars something containing a directory and then tries to access it, e.g. copying something inside it, reading something from it, etc.? It sounds like that would fail without this fix?

Yes, it would fail without the fix.
The existing tests have the expected permission hard coded, so they'll also fail if this regresses.

**Review by stephentoub — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]
:shipit:

**Review by carlossanlop — APPROVED** [review]
LGTM. I ran the `runtime-extra-platforms` pipeline, to make sure all the exotic platforms pass too. 
If it all looks good, we can merge it.

**Comment by carlossanlop** [comment]
No tar failures. :shipit:

**Pull request merged**